### PR TITLE
[Python Distribution] Add a check for file size

### PR DESCRIPTION
**Pull request by XuanWang-Amos**
Add a step in distribution test which checks if any of the artifacts exceeds 80 MB since Pypi has a per-file size limit.

### CODE REVIEW

File: tools/internal_ci/helper_scripts/check_python_artifacts_size.sh
Comment (gnossen) [inline tools/internal_ci/helper_scripts/check_python_artifacts_size.sh:21]: Since the artifacts are currently over 80MB, we have two options:

1. Temporarily bump this to 110MB and have a TODO (which we actually follow up on) or
2. Wait to merge this PR until we actually fix the file size.

It's up to you which one to go with.

**Review by gnossen — APPROVED** [review]

File: tools/internal_ci/helper_scripts/check_python_artifacts_size.sh
Comment (XuanWang-Amos) [reply tools/internal_ci/helper_scripts/check_python_artifacts_size.sh:21]: Since this PR won't be useful until next release, I'll merge it after we fix the size issue.

**Comment by gnossen** [comment]
One half of the long-term solution for the mitigation provided by https://github.com/pypi/support/issues/3616

**Comment by gnossen** [comment]
Now that https://github.com/grpc/grpc/pull/35832 is in, this should be good to go.

**Pull request closed without merging**
